tests/integration/contract-call/locked-wallet.spec.ts

```diff
@@ -0,0 +1,72 @@
+import { RouteUrls } from '@shared/route-urls';
+
+import { SendPage } from '../../page-objects/send-form.page';
+import { WalletPage } from '../../page-objects/wallet.page';
+import { BrowserDriver, createTestSelector, selectTestnet, setupBrowser } from '../utils';
+import { SECRET_KEY_2 } from '@tests/mocks';
+import { SettingsSelectors } from '@tests/integration/settings.selectors';
+import { Page } from 'playwright-core';
+
+jest.setTimeout(60_0000);
+jest.retryTimes(process.env.CI ? 2 : 0);
+
+describe('Locked wallet test', () => {
+  const BEFORE_EACH_TIMEOUT = 600000;
+
+  let browser: BrowserDriver;
+  let walletPage: WalletPage;
+  let mainPage: WalletPage;
+  let sendForm: SendPage;
+  let pages: Page[];
+  let latestPage: Page;
+
+  beforeEach(async () => {
+    browser = await setupBrowser();
+    walletPage = await WalletPage.init(browser, RouteUrls.Onboarding);
+    await walletPage.signIn(SECRET_KEY_2);
+    await walletPage.waitForHomePage();
+    // switch to test net
+    await selectTestnet(walletPage);
+    pages = await WalletPage.getAllPages(browser);
+    mainPage = new WalletPage(pages[1]);
+    await mainPage.clickSignUp();
+
+    await walletPage.page.waitForTimeout(2000);
+    pages = await WalletPage.getAllPages(browser);
+    latestPage = pages[pages.length - 1];
+    sendForm = new SendPage(latestPage);
+    await sendForm.waitForPreview('$account1');
+    await sendForm.clickFirstAccount();
+  }, BEFORE_EACH_TIMEOUT);
+
+  afterEach(async () => {
+    try {
+      await browser.context.close();
+    } catch (error) {}
+  });
+
+  it('when wallet is not locked', async () => {
+    await mainPage.clickContractCall();
+    await walletPage.page.waitForTimeout(2000);
+    pages = await WalletPage.getAllPages(browser);
```

Comment:
You set `pages` already in the `beforeEach` block. Why do we need to do it again? Is the result different?

Reply:
Yeah i'm not 100% sure but every test needs it to pass.
This is the failure I get when I remove it.
Would you keep it as is or try to change it and what would be the suggested change.


```

 FAIL  tests/integration/contract-call/locked-wallet.spec.ts (42.35 s)
  Locked wallet test
    ✕ when wallet is not locked (18603 ms)
    ✓ when wallet is locked it shows unlock screen (23141 ms)

  ● Locked wallet test › when wallet is not locked

    page.waitForSelector: Target closed
    =========================== logs ===========================
    waiting for selector "[data-testid="standard-fee"]" to be visible
    ============================================================

      166 |
      167 |   async waitForPreview(selector: keyof typeof selectors) {
    > 168 |     await this.page.waitForSelector(this.selectors[selector]);
          |                     ^
      169 |   }
      170 |
      171 |   async inputFee(input: string) {

      at SendPage.waitForPreview (tests/page-objects/send-form.page.ts:168:21)
      at Object.<anonymous> (tests/integration/contract-call/locked-wallet.spec.ts:54:20
```

Reply:
sure